src/commands/list-kinesis-streams.js

```diff
@@ -0,0 +1,235 @@
+const _ = require("lodash");
+const { getAWSSDK } = require("../lib/aws");
+const Table = require("cli-table");
+const { Command, flags } = require("@oclif/command");
+const { checkVersion } = require("../lib/version-check");
+const Kinesis = require("../lib/kinesis");
+require("colors");
+const uuid = require("uuid");
+const { getStreamsInRegion } = require("../lib/kinesis");
+const { getRegionFromARN } = require("../lib/aws");
+
+const FIVE_MIN_IN_SECONDS = 60 * 5;
+const MAX_RECORDS_PER_SHARD_WRITE = 1000;
+const MAX_RECORDS_PER_SHARD_READ = 2000;
+const BYTES_INCOMING_PER_SHARD = 1024 * 1024;
+const BYTES_OUTGOING_PER_SHARD = 2048 * 1024;
+
+const formatFloat = num => {
+	return Number.parseFloat(num).toFixed(2);
+};
+
+class ListKinesisStreamsCommand extends Command {
+	async run() {
+		const { flags } = this.parse(ListKinesisStreamsCommand);
+		const { region, profile } = flags;
+
+		global.profile = profile;
+
+		checkVersion();
+		let streamsDescription = [];
+		if (region) {
+			this.log(`Checking Kinesis streams in [${region}]`);
+			streamsDescription = await getStreamsDescriptionFromRegion(region);
+		} else {
+			this.log("Checking Kinesis streams in all regions");
+			streamsDescription = await getStreamsDescriptionFromAllRegions();
+		}
+		show(streamsDescription);
+	}
+}
+
+ListKinesisStreamsCommand.description = "Lists the Kinesis streams";
+ListKinesisStreamsCommand.flags = {
+	region: flags.string({
+		char: "r",
+		description: "AWS region, e.g. us-east-1",
+		required: false
+	}),
+	profile: flags.string({
+		char: "p",
+		description: "AWS CLI profile name",
+		required: false
+	})
+};
+
+const getStreamsDescriptionFromRegion = async region => {
+	let streamNames = await getStreamsInRegion(region);
+
+	const usageStats = await getUsageMetrics(streamNames, region);
+	return describeStreams(streamNames, region, usageStats);
+};
+
+const getStreamsDescriptionFromAllRegions = async () => {
+	const promises = Kinesis.regions.map(region =>
+		getStreamsDescriptionFromRegion(region)
+	);
+	const results = await Promise.all(promises);
+	return _.flatMap(results);
+};
+
+const describeStreams = async (streams, region, usageStats) => {
+	const streamsDescription = [];
+	for (const stream of streams) {
+		streamsDescription.push(describeStream(stream, region, usageStats[stream]));
+	}
```

Comment:
not an issue but it's probably nicer as a `map`

Reply:
I'll ignore gracefully ;-)